### PR TITLE
feat: Add passwd session mode

### PR DESCRIPTION
**Pull request by denisonbarbosa**
We used to only support authentication sessions (coming from `pam_sm_authenticate`). Now, we also intend to support password changes (coming from `pam_sm_chauthtok`). To do that, we need to expand the API to allow the creation of different session types.

UDENG-2048

### CODE REVIEW

**Comment by denisonbarbosa** [comment]
I created another card for updating the ExampleBroker and the PAM module as it will require quite more work and we should write a different test workflow, i.e. have TestCLIAuthIntegration and TestCLIPasswdIntegration, IMHO.

File: internal/brokers/examplebroker/broker.go [hunk 1]
```diff
@@ -36,9 +36,11 @@ const (
 )
 
 type sessionInfo struct {
-	username        string
-	selectedMode    string
-	lang            string
+	username    string
+	lang        string
+	sessionMode string
+
```
Comment (didrocks) [inline internal/brokers/examplebroker/broker.go:42]: I was going to debate on the line break, but reading through, you convinced me :)

File: authd.proto [hunk 2]
```diff
@@ -45,6 +45,7 @@ message SBRequest {
   string broker_id = 1;
   string username = 2;
   string lang = 3;
+  string mode = 4;
```
Comment (didrocks) [inline authd.proto:48]: Maybe this is one place where we can use an enum, wdyt?

**Review by didrocks — COMMENTED** [review]
Excellent work! I have one small food for thoughts, but all the rest, including tests adaptation, looks gorgeous!

I agree in splitting the task in 2 cards.

File: authd.proto (hunk 2 above)
Comment (denisonbarbosa) [reply authd.proto:48]: Sounds good to me. This way, we kind of avoid the whole "not defined" issue.

File: authd.proto [hunk 3]
```diff
@@ -41,11 +41,17 @@ message StringResponse {
   string msg = 1;
 }
 
+enum SessionMode {
+  UNDEFINED = 0;
```
Comment (didrocks) [inline authd.proto:45]: Why having UNDEFINED? I think it’s better, as this is a Go project that the 0 value is useful, and so, default to authd.

**Review by didrocks — CHANGES_REQUESTED** [review]
just a nitpick about the zero value!

File: authd.proto (hunk 3 above)
Comment (denisonbarbosa) [reply authd.proto:45]: I was wondering about the same thing. I looked at other enum examples in gRPC and it seemed to me as though having an undefined value is idiomatic in some ways.

Comment (didrocks) [reply authd.proto:45]: Yeah, in general in APIs, that’s the way, but here, we are from Go to Go, so I would say the idiomatic way is the Go one? (also, there is arguably a main mode, which is auth)

Comment (denisonbarbosa) [reply authd.proto:45]: Also, one of the protobuf strengths is the multi-language compatibility so, even though we are using this in a Go project, we should not restrict it to it. Wdyt?

Comment (didrocks) [reply authd.proto:45]: Ack, happy to give it a pass :)

**Review by didrocks — APPROVED** [review]

**Pull request merged**